packages/patapata_core/lib/src/network.dart

```diff
@@ -50,32 +50,34 @@ enum NetworkConnectivity {
 /// This class, within [NetworkPlugin]
 /// obtains the network status using [Connectivity].
 class NetworkInformation {
-  final NetworkConnectivity connectivity;
+  final List<NetworkConnectivity> connectivities;
 
   const NetworkInformation({
-    required this.connectivity,
+    required this.connectivities,
   });
 
   factory NetworkInformation.unknown() => const NetworkInformation(
-        connectivity: NetworkConnectivity.unknown,
+        connectivities: [NetworkConnectivity.unknown],
       );
 
   @override
-  String toString() => 'NetworkInformation:connectivity=${connectivity.name}';
+  String toString() => 'NetworkInformation:connectivity=[${[
+        for (final tConnectivity in connectivities) tConnectivity.name
+      ]}]';
 
   NetworkInformation copyWith({
-    NetworkConnectivity? connectivity,
+    List<NetworkConnectivity>? connectivities,
   }) =>
       NetworkInformation(
-        connectivity: connectivity ?? this.connectivity,
+        connectivities: connectivities ?? this.connectivities,
       );
 
   @override
   operator ==(Object other) =>
-      other is NetworkInformation && connectivity == other.connectivity;
+      other is NetworkInformation && connectivities == other.connectivities;
 
   @override
-  int get hashCode => Object.hash('NetworkInformation', connectivity);
+  int get hashCode => Object.hash('NetworkInformation', connectivities);
```

Comment:
This is incorrect. This will always be false since you're now comparing two Lists. Before it was checking a single object.

You'll have do a deep compare on the list to see if they are the same or not.